include/rtc/rtp.hpp

```diff
@@ -0,0 +1,521 @@
+    uint32_t timestamp;
+    SSRC ssrc;
+    SSRC csrc[16];
+
+    inline uint32_t getSeqNo() const {
+        return ntohs(seqNumber);
+    }
+
+    inline uint32_t getTS() const{
+        return ntohl(timestamp);
+    }
+};
+
+ struct RTCP_REMB
+{
+     RTCP_HEADER header;
+
+    SSRC senderSSRC;
+    SSRC mediaSourceSSRC;
+
+    /*! \brief Unique identifier ('R' 'E' 'M' 'B') */
+    char id[4];
+
+    /*! \brief Num SSRC, Br Exp, Br Mantissa (bit mask) */
+    uint32_t bitrate;
+
+    SSRC ssrc[1];
+
+    [[nodiscard]] unsigned int getSize() const {
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        return sizeof(uint32_t) * (1+header.getLength());
+    }
+
+    void preparePacket(SSRC senderSSRC, unsigned int numSSRC, unsigned int bitrate) {
+//        version = 2;
+//        format = 15;
+//        padding = false;
+//        payloadType = 206;
+
+        // Report Count becomes the format here.
+        header.prepareHeader(206, 15, 0);
+
+        // Always zero.
+        mediaSourceSSRC = 0;
+
+        this->senderSSRC = htonl(senderSSRC);
+        id[0] = 'R';
+        id[1] = 'E';
+        id[2] = 'M';
+        id[3] = 'B';
+
+        setBitrate(numSSRC, bitrate);
+    }
+
+    void setBitrate(unsigned int numSSRC, unsigned int bitrate) {
+        unsigned int exp = 0;
+        while (bitrate > pow(2,18)-1) {
+            exp++;
+            bitrate /= 2;
+        }
+
+        // "length" in packet is one less than the number of 32 bit words in the packet.
+        header.setLength((offsetof(RTCP_REMB, ssrc)/4)-1+numSSRC);
+
+        this->bitrate = htonl(
+                (numSSRC << (32u-8u)) | (exp << (32u-8u-6u)) | bitrate
+        );
+    }
+
+    //TODO Make this work
+//    uint64_t getBitrate() const{
+//        uint32_t ntohed = ntohl(this->bitrate);
+//        uint64_t bitrate = ntohed & (unsigned int)(pow(2, 18)-1);
+//        unsigned int exp = ntohed & ((unsigned int)( (pow(2, 6)-1)) << (32u-8u-6u));
+//        return bitrate * pow(2,exp);
+//    }
+//
+//    uint8_t getNumSSRCS() const {
+//        return ntohl(this->bitrate) & (((unsigned int) pow(2,8)-1) << (32u-8u));
+//    }
+
+    void print() {
+        std::cout << "REMB ";
+        header.print();
+        std::cout << " SSRC:" << ntohl(senderSSRC);
+    }
+
+    void setSSRC(uint8_t iterator, SSRC ssrc) {
+        this->ssrc[iterator] = htonl(ssrc);
+    }
+
+    static unsigned int sizeWithSSRCs(int numSSRC) {
+        return (offsetof(RTCP_REMB, ssrc))+sizeof(SSRC)*numSSRC;
+    }
+};
+
+class RTCPSession {
+private:
+    std::function<void(RTP)> onPacketCB;
+    unsigned int requestedBitrate = 0;
+//    synchronized_callback<const rtc::message_ptr&> txCB;
+    std::function<void(rtc::message_ptr)> txCB;
+    SSRC ssrc;
+    uint32_t greatestSeqNo = 0, greatestTS;
+    uint64_t syncRTPTS, syncNTPTS;
+
+    unsigned int rotationCount = 0;
+
+
+public:
+    std::vector<rtc::message_ptr> sendQueue;
```

Comment:
There is a dedicated thread-safe message queue that you could use in `include/rtc/queue.hpp`.